ethexe/contracts/src/Router.sol

```diff
@@ -379,34 +377,51 @@ contract Router is IRouter, OwnableUpgradeable, ReentrancyGuardTransient {
 
     /* Helper private functions */
 
-    function _createProgram(bytes32 _codeId, bytes32 _salt) private returns (address) {
-        Storage storage router = _router();
+    function _createProgram(bytes32 _codeId, Storage storage router) private returns (address program) {
         require(router.genesisBlock.hash != bytes32(0), "router genesis is zero; call `lookupGenesisHash()` first");
 
         require(
             router.protocolData.codes[_codeId] == Gear.CodeState.Validated,
             "code must be validated before program creation"
         );
 
-        // Check for duplicate isn't necessary, because `Clones.cloneDeterministic`
-        // reverts execution in case of address is already taken.
-        address actorId =
-            Clones.cloneDeterministic(router.implAddresses.mirrorProxy, keccak256(abi.encodePacked(_codeId, _salt)));
```

Comment:
we can keep `create2` and `cloneDeterministic` for raw programs (interactions with bytes, not with interface)